pyresample/gradient/__init__.py

```diff
@@ -33,25 +36,49 @@
 from shapely.geometry import Polygon
 
 from pyresample import CHUNK_SIZE
-from pyresample.geometry import get_geostationary_bounding_box
-from pyresample.gradient._gradient_search import one_step_gradient_search
-from pyresample.resampler import BaseResampler
+from pyresample.geometry import (
+    AreaDefinition,
+    SwathDefinition,
+    get_geostationary_bounding_box,
+)
+from pyresample.gradient._gradient_search import (
+    one_step_gradient_indices,
+    one_step_gradient_search,
+)
+from pyresample.resampler import BaseResampler, resample_blocks
 
 logger = logging.getLogger(__name__)
 
 
+def GradientSearchResampler(source_geo_def, target_geo_def):
+    """Create a gradient search resampler."""
+    warnings.warn("`GradientSearchResampler` is deprecated, please use "
+                  "`create_gradient_search_resampler` instead.",
+                  DeprecationWarning)
+    return create_gradient_search_resampler(source_geo_def, target_geo_def)
+
+
+def create_gradient_search_resampler(source_geo_def, target_geo_def):
+    """Create a gradient search resampler."""
+    if isinstance(source_geo_def, AreaDefinition) and isinstance(target_geo_def, AreaDefinition):
+        return ResampleBlocksGradientSearchResampler(source_geo_def, target_geo_def)
+    elif isinstance(source_geo_def, SwathDefinition) and isinstance(target_geo_def, AreaDefinition):
+        return StackingGradientSearchResampler(source_geo_def, target_geo_def)
+    raise NotImplementedError
+
+
 @da.as_gufunc(signature='(),()->(),()')
 def transform(x_coords, y_coords, src_prj=None, dst_prj=None):
     """Calculate projection coordinates."""
     return pyproj.transform(src_prj, dst_prj, x_coords, y_coords)
 
 
-class GradientSearchResampler(BaseResampler):
+class StackingGradientSearchResampler(BaseResampler):
```

Comment:
The docstring is identical to `ResampleBlocksGradientSearchResampler`, what's the difference between these?

Reply:
Improved the docstrings.